src/runtime.js

```diff
@@ -151,6 +151,7 @@ let tornado = {
    */
   exists(val, placeholderNode, bodies, context) {
     if (this.util.isPromise(val)) {
+      placeholderNode = this.insertPendingBody(placeholderNode, bodies.pending, context) || placeholderNode;
```

Comment:
agree we should optimize the common usecase. but as an implementation `exists` and `section` should be no different then any other helper. We should build them on top of the same APIs helper authors use. 

It'll be nice to have a few APIs for the common use cases:
* extend built-ins
* work like how filters are being used today
* create new DOM nodes
* modify a copy of context.

etc.